Contributor guide: add info about gitlab repository for data (#4892)
scikit-image/scikit-image: CONTRIBUTING.txt

     - ``upstream``, which refers to the ``scikit-image`` repository
     - ``origin``, which refers to your personal fork

.. note::

    Although our code is hosted on `github

>     <https://github.com/scikit-image/>`_, our database is stored on `gitlab

Reviewer: ```suggestion
    <https://github.com/scikit-image/>`_, our datasets are stored on `gitlab
```

Reviewer: database sounds like we have a big database like mysql etc.

Reviewer 2: @sciunto you've marked this as resolved but I still see database?

>     <https://github.com/scikit-image/>`_, our database is stored on `gitlab

Author: ```suggestion
    <https://github.com/scikit-image/>`_, our dataset is stored on `gitlab
```

>     <https://gitlab.com/scikit-image>`_. and fetched with `pooch

Reviewer 2: Add full link? I think it's

```suggestion
    <https://gitlab.com/scikit-image/data>`_ and fetched with `pooch
```
